src/components/duelling-picklist/picklist-group/picklist-group.component.tsx

```diff
@@ -17,9 +16,38 @@ import {
 import FocusContext from "../duelling-picklist.context";
 import Events from "../../../__internal__/utils/helpers/events";
 
-const PicklistGroup = React.forwardRef(
+export interface PicklistGroupProps {
+  /** Group title */
+  title: React.ReactNode;
+  /** Item content */
+  children: React.ReactNode;
+  /** Define if item is of type add or remove */
+  type: "add" | "remove";
+  /** Handler invoked when add/remove button is clicked or when space/enter is pressed on the whole item */
+  onChange: () => void;
+  /** @private @ignore */
+  index?: number;
+  /** @private @ignore */
+  listIndex?: number;
+  /** @private @ignore */
+  isLastGroup?: boolean;
+}
+
+export const PicklistGroup = React.forwardRef<
+  HTMLButtonElement,
+  PicklistGroupProps
+>(
   (
-    { title, children, type, onChange, index, listIndex, isLastGroup, ...rest },
+    {
+      title,
+      children,
+      type,
+      onChange,
+      index,
+      listIndex,
+      isLastGroup,
+      ...rest
```

Comment:
It seems that it actually has to stay as it is not an empty object. The PicklistItem components are being rendered as a direct child of TransitionGroup meaning that there are CSSTransition specific props injected into this component.

I've renamed rest to transitionGroupProps though so that it's obvious why it's there

Reply:
that's a bit confusing if `CSSTransition` is somehow injecting props that are not allowed by the props interface. I guess it's a JS component them? (Not ours so we don't have to worry about it, but it's hard to avoid coming to the same incorrect conclusion James came to when looking at this code. The renaming helps though!)